src/install/dracut-install.c

```diff
@@ -183,9 +183,7 @@ static char *convert_abs_rel(const char *from, const char *target)
         target_dir_p = strndup(target, dirlen);
         if (!target_dir_p)
                 return strdup(from + strlen(destrootdir));
-
-        realpath_p = realpath(target_dir_p, NULL);
-        if (realpath_p == NULL) {
+        if (realpath(target_dir_p, relative_from) == NULL) {
```

Comment:
`relative_from` as a temporary buffer to build `realfrom` and `realtarget` is fine, but its name is confusing. Maybe rename it to `buf` or whatever? (comments below will need update in this case)